src/overlays/kaleido_scope/ovl_kaleido_scope/z_kaleido_mask.c

```diff
@@ -1,59 +1,810 @@
+    }
+
+    CLOSE_DISPS(play->state.gfxCtx);
+}
+
+u8 sMaskPlayerFormItems[] = {
+    ITEM_MASK_FIERCE_DEITY, ITEM_MASK_GORON, ITEM_MASK_ZORA, ITEM_MASK_DEKU, ITEM_NONE,
 };
 
-s16 D_8082B6E4 = 0;
+void KaleidoScope_UpdateMaskCursor(PlayState* play) {
+    Input* input = CONTROLLER1(&play->state);
+    PauseContext* pauseCtx = &play->pauseCtx;
+    MessageContext* msgCtx = &play->msgCtx;
+    u16 vtxIndex;
+    u16 cursorItem;
+    u16 cursorSlot;
+    s16 cursorPoint;
+    s16 cursorXIndex;
+    s16 cursorYIndex;
+    s16 oldCursorPoint;
+    s16 moveCursorResult;
+    s16 pad2;
+
+    pauseCtx->cursorColorSet = 0;
+    pauseCtx->nameColorSet = 0;
+
+    if ((pauseCtx->state == 6) && (pauseCtx->unk_200 == 0) && (pauseCtx->pageIndex == PAUSE_MASK) &&
+        !pauseCtx->itemDescriptionOn) {
+        moveCursorResult = 0;
+        oldCursorPoint = pauseCtx->cursorPoint[PAUSE_MASK];
+
+        cursorItem = pauseCtx->cursorItem[PAUSE_MASK];
+
+        // Move cursor left/right
+        if (pauseCtx->cursorSpecialPos == 0) {
+            // cursor is currently on a slot
+            pauseCtx->cursorColorSet = 2;
+
+            if (ABS_ALT(pauseCtx->stickAdjX) > 30) {
+                cursorPoint = pauseCtx->cursorPoint[PAUSE_MASK];
+                cursorXIndex = pauseCtx->cursorXIndex[PAUSE_MASK];
+                cursorYIndex = pauseCtx->cursorYIndex[PAUSE_MASK];
+
+                // Search for slot to move to
+                while (moveCursorResult == 0) {
+                    if (pauseCtx->stickAdjX < -30) {
+                        // move cursor left
+                        pauseCtx->unk_298 = 4.0f;
+                        if (pauseCtx->cursorXIndex[PAUSE_MASK] != 0) {
+                            pauseCtx->cursorXIndex[PAUSE_MASK]--;
+                            pauseCtx->cursorPoint[PAUSE_MASK]--;
+                            moveCursorResult = 1;
+                        } else {
+                            pauseCtx->cursorYIndex[PAUSE_MASK]++;
+
+                            if (pauseCtx->cursorYIndex[PAUSE_MASK] >= 4) {
+                                pauseCtx->cursorYIndex[PAUSE_MASK] = 0;
+                            }
+
+                            pauseCtx->cursorPoint[PAUSE_MASK] =
+                                pauseCtx->cursorXIndex[PAUSE_MASK] + (pauseCtx->cursorYIndex[PAUSE_MASK] * 6);
+
+                            if (pauseCtx->cursorPoint[PAUSE_MASK] >= NUM_MASK_SLOTS) {
+                                pauseCtx->cursorPoint[PAUSE_MASK] = pauseCtx->cursorXIndex[PAUSE_MASK];
+                            }
+
+                            if (cursorYIndex == pauseCtx->cursorYIndex[PAUSE_MASK]) {
+                                pauseCtx->cursorXIndex[PAUSE_MASK] = cursorXIndex;
+                                pauseCtx->cursorPoint[PAUSE_MASK] = cursorPoint;
+
+                                KaleidoScope_MoveCursorToSpecialPos(play, PAUSE_CURSOR_PAGE_LEFT);
+
+                                moveCursorResult = 2;
```

Comment:
Can `moveCursorResult` be an enum? It seems like it has a few magic values. I might be coming in late on this discussion, since `z_kaleido_item` is already in.

Reply:
I took a quick look at all the kaleido pages, and it seems this system of using a `moveCursorResult` is only for the mask and item page, and it's probably only part of the mask page because it was copied over from OoT. I think there could be an enum, but it's kinda local to this function. But because of copied code, it's in two different files. So I'm not sure if I should have two identical enums above the function, or 1 shared enum in kaleido.h. Thoughts?

Reply:
I think one shared enum (with maybe a comment above it) would be fine.